core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -2109,35 +1898,147 @@ func (o *evmTxStore) UpdateTxAttemptBroadcastBeforeBlockNum(ctx context.Context,
 	return err
 }
 
-// FindConfirmedTxesReceipts Returns all confirmed transactions with receipt block nums older than or equal to the finalized block number
-func (o *evmTxStore) FindConfirmedTxesReceipts(ctx context.Context, finalizedBlockNum int64, chainID *big.Int) (receipts []Receipt, err error) {
+// FindAttemptsRequiringReceiptFetch returns all broadcasted attempts for confirmed or terminally stuck transactions that do not have receipts stored in the DB
+func (o *evmTxStore) FindAttemptsRequiringReceiptFetch(ctx context.Context, chainID *big.Int) (attempts []TxAttempt, err error) {
+	var cancel context.CancelFunc
+	ctx, cancel = o.stopCh.Ctx(ctx)
+	defer cancel()
+	var dbTxAttempts []DbEthTxAttempt
+	query := `
+		SELECT evm.tx_attempts.* FROM evm.tx_attempts
+		JOIN evm.txes ON evm.txes.ID = evm.tx_attempts.eth_tx_id
+		WHERE evm.tx_attempts.state = 'broadcast' AND evm.txes.state IN ('confirmed', 'confirmed_missing_receipt', 'fatal_error') AND evm.txes.evm_chain_id = $1 AND evm.txes.ID NOT IN (
+			SELECT DISTINCT evm.txes.ID FROM evm.txes
+			JOIN evm.tx_attempts ON evm.tx_attempts.eth_tx_id = evm.txes.ID
+			JOIN evm.receipts ON evm.receipts.tx_hash = evm.tx_attempts.hash
+			WHERE evm.txes.evm_chain_id = $1 AND evm.txes.state IN ('confirmed', 'confirmed_missing_receipt', 'fatal_error') AND evm.receipts.ID IS NOT NULL
+		)
```

Comment:
I am curious if we can use another join to replace the `NOT IN (....)`, since the subquery seems a bit complicated

Reply:
Dimitris suggested the same earlier but I wasn't able to simplify it with another join. Don't think I can just add JOIN on the receipt table with a `evm.receipts.ID IS NOT NULL` condition. That would still include attempts for transactions that have receipts. But if we find during testing this query is too heavy, I could consider it making the change and be ok with the extra RPC calls since it's in a batch anyways.